examples/threadExample/src/main/java/io/realm/examples/threads/AsyncQueryFragment.java

```diff
@@ -0,0 +1,193 @@
+                .between("y", 0, 50)
+                .findAllSortedAsync(
+                         "x", RealmResults.SORT_ORDER_ASCENDING,
+                         "y", RealmResults.SORT_ORDER_DESCENDING
+                 );
+        dotAdapter.updateList(allSortedDots);
+        allSortedDots.addChangeListener(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        // Remember to close the Realm instance when done with it.
+        cancelTransaction();
+        allSortedDots.removeChangeListener(this);
+        allSortedDots = null;
+        realm.close();
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.translate_button: {
+                cancelTransaction();
+                // translate all points coordinates using an async transaction
+                asyncTransaction = realm.executeTransaction(new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        // query for all points
+                        RealmResults<Dot> dots = realm.where(Dot.class).findAll();
+
+                        for (int i = dots.size() - 1; i >= 0; i--) {
+                            Dot dot = dots.get(i);
+                            if (dot.isValid()) {
+                                int x = dot.getX();
+                                int y = dot.getY();
+                                dot.setX(y);
+                                dot.setY(x);
+                            }
+                        }
+                    }
+                }, new Realm.Transaction.Callback() {
+                    @Override
+                    public void onSuccess() {
+                        if (isAdded()) {
+                            Toast.makeText(getActivity(), "Translation completed", Toast.LENGTH_SHORT).show();
+                        }
+                    }
+
+                    @Override
+                    public void onError(Exception e) {
+                        if (isAdded()) {
+                            Toast.makeText(getActivity(), "Error while translating dots", Toast.LENGTH_SHORT).show();
+                            e.printStackTrace();
+                        }
+                    }
+                });
+                break;
+            }
+        }
+    }
+
+    private void cancelTransaction() {
+        if (asyncTransaction != null && !asyncTransaction.isCancelled()) {
+            asyncTransaction.cancel();
+            asyncTransaction = null;
+        }
+    }
+
+    @Override
+    public void onChange() {
+        dotAdapter.notifyDataSetChanged();
+    }
+
+
+    // using a generic Adapter instead of RealmBaseAdapter, because
+    // RealmBaseAdapter register a listener against all Realm changes
+    // whereas in this scenario we're only interested on the changes of our query
+    private class DotAdapter extends BaseAdapter {
```

Comment:
Why not private static? It doesn't use any methods from the enclosing class as far as I can see?